src/optimizer/plan_generator.cpp

```diff
@@ -52,25 +52,28 @@
 
 namespace noisepage::optimizer {
 
-PlanGenerator::PlanGenerator() = default;
+PlanGenerator::PlanGenerator(common::ManagedPointer<planner::PlanMetaData> plan_meta_data)
+    : plan_id_counter_(0), plan_meta_data_(plan_meta_data) {}
 
 std::unique_ptr<planner::AbstractPlanNode> PlanGenerator::ConvertOpNode(
     transaction::TransactionContext *txn, catalog::CatalogAccessor *accessor, AbstractOptimizerNode *op,
     PropertySet *required_props, const std::vector<common::ManagedPointer<parser::AbstractExpression>> &required_cols,
     const std::vector<common::ManagedPointer<parser::AbstractExpression>> &output_cols,
-    std::vector<std::unique_ptr<planner::AbstractPlanNode>> &&children_plans,
-    std::vector<ExprMap> &&children_expr_map) {
+    std::vector<std::unique_ptr<planner::AbstractPlanNode>> &&children_plans, std::vector<ExprMap> &&children_expr_map,
+    const planner::PlanMetaData::PlanNodeMetaData &plan_node_meta_data) {
   required_props_ = required_props;
   required_cols_ = required_cols;
   output_cols_ = output_cols;
   children_plans_ = std::move(children_plans);
   children_expr_map_ = children_expr_map;
   accessor_ = accessor;
+  plan_node_meta_data_ = plan_node_meta_data;
```

Comment:
Here you can use `plan_node_meta_data_ = std::move(plan_node_meta_data);` to save the copy.

Reply:
Clang-tidy does not allow this.